File: ios/Releaf/Features/Notepad/Recents/Views/TagChips.swift
import SwiftUI
import ReleafDesignSystem

/// Horizontal scrollable strip of tag-filter chips. Selected = green800 / textOnDark.
struct TagChips: View {
    @Binding var selection: TagFilter

    private let filters: [TagFilter] = [
        .all,
        .tag(.home),
        .tag(.work),
        .tag(.recipes),
        .tag(.personal)
    ]

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 8) {
                ForEach(Array(filters.enumerated()), id: \.offset) { _, filter in
                    chip(for: filter)
                }
            }
            .padding(.horizontal, 2)
        }
    }

    @ViewBuilder
    private func chip(for filter: TagFilter) -> some View {
        let isSelected = filter == selection
        Button {
            selection = filter
        } label: {
            Text(filter.label)
                .font(.system(size: 13, weight: isSelected ? .medium : .regular))
                .foregroundColor(isSelected ? .textOnDark : .textGreen)
                .padding(.horizontal, 14)
                .padding(.vertical, 8)
                .background(
                    Capsule(style: .continuous)
                        .fill(isSelected ? Color.green800 : Color.bgChip)
                )
        }
        .buttonStyle(.plain)
    }
}

#if DEBUG
private struct TagChipsPreview: View {
    @State var sel: TagFilter = .all
    var body: some View {
        TagChips(selection: $sel)
            .padding()
            .background(Color.bgCanvas)
    }
}
#Preview { TagChipsPreview() }
#endif

File: android/app/src/main/java/app/releaf/mobile/features/notepad/recents/ui/TagChips.kt
package app.releaf.mobile.features.notepad.recents.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import app.releaf.mobile.features.notepad.recents.model.Tag
import app.releaf.mobile.features.notepad.recents.model.TagFilter
import app.releaf.mobile.features.notepad.recents.theme.BgChip
import app.releaf.mobile.features.notepad.recents.theme.Green800
import app.releaf.mobile.features.notepad.recents.theme.TextOnDark
import app.releaf.mobile.features.notepad.recents.theme.Type

/** Horizontal scrolling chip row: All / Home / Work / Recipes / Personal. */
@Composable
fun TagChips(
    selected: TagFilter,
    onSelected: (TagFilter) -> Unit,
    modifier: Modifier = Modifier,
) {
    val items: List<TagFilter> = buildList {
        add(TagFilter.All)
        addAll(Tag.values().map { TagFilter.Single(it) })
    }

    LazyRow(
        modifier = modifier,
        contentPadding = PaddingValues(horizontal = 14.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        items(items) { item ->
            val isActive = item == selected
            Chip(
                label = item.label(),
                isActive = isActive,
                onClick = { onSelected(item) },
            )
        }
    }
}

private fun TagFilter.label(): String = when (this) {
    is TagFilter.All -> "All"
    is TagFilter.Single -> tag.display
}

@Composable
private fun Chip(
    label: String,
    isActive: Boolean,
    onClick: () -> Unit,
) {
    val bg = if (isActive) Green800 else BgChip
    val fg = if (isActive) TextOnDark else Green800
    // Brief: only Regular and Medium. Active = Medium for emphasis.
    val weight = if (isActive) FontWeight.Medium else FontWeight.Normal

    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(999.dp))
            .background(bg)
            .clickable(onClick = onClick)
            .padding(horizontal = 14.dp, vertical = 8.dp),
        contentAlignment = Alignment.Center,
    ) {
        BasicText(
            text = label,
            style = Type.Chip.copy(color = fg, fontWeight = weight),
        )
    }
}
